### PR TITLE
Add test

### PR DESCRIPTION
**Pull request by ademibowale**
## In this project 
# Changes have been made according to my code reviewer requested changes
- Added more integration test for the post and the user .
- Testing for the post
- Testing for the user

Added the required changes suggested by @AnselemOdims in the post_spec.rb in the features folder

### CODE REVIEW

File: spec/intergrationTests/post_index_page_spec.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    it 'I can see how many comments the post has' do
+      visit user_posts_path(@user)
+      expect(page).to have_content('Comments 0')
+    end
+
+    it 'I can see section of pagination if there' do
+      visit user_posts_path(@user)
+      expect(page).not_to have_content('Pagination')
+    end
+
```
Comment (alexander16108) [inline spec/intergrationTests/post_index_page_spec.rb]: as required there is still a test case required for this post index page file

- Tests that asserts you can see how many likes a post has.
-

File: spec/intergrationTests/post_show_page_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      visit user_post_path(@user, @post)
+      expect(page).to have_content('This is a post')
+    end
+
+    it 'display the posts comments' do
+      visit user_post_path(@user, @post)
+      expect(page).to have_content('Comments')
+    end
+  end
+end
```
Comment (alexander16108) [inline spec/intergrationTests/post_show_page_spec.rb]: as required there is still a test case required for this post-index page file

- Tests that asserts you can see the username of each commentor.

File: spec/intergrationTests/users_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    it 'I can see button to see all posts' do
+      visit user_path(@user)
+      expect(page).to have_link('See all posts')
+    end
+
+    it 'when I click a user\s post I am redirected to the post show page' do
+      visit user_path(@user)
+      click_link('See all posts')
+      expect(page).to have_current_path(user_posts_path(@user))
+    end
```
Comment (alexander16108) [inline spec/intergrationTests/users_show_spec.rb]: please add  the required test that 
- Tests that redirects to the user's post's index page when you click on a **see all posts**.

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 


 ## Required Changes ♻️
_Check the comments under the review._


the following test cases are not available for the User index page add them and try again
* Spec file should only test `User` index page. It can be only one spec testing all off the things below or more, ideally one for the asserts and one for the redirect
* Tests that asserts you can see the username of all other users.
* Tests that asserts you can see the profile picture for each user.
* Tests that asserts you can see the number of posts each user has written.
* Tests that redirects to the user's show page when you click on a user.


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/intergrationTests/users_show_spec.rb (hunk 3 above)
Comment (ademibowale) [reply spec/intergrationTests/users_show_spec.rb]: Thank you my great developer

File: spec/intergrationTests/post_show_page_spec.rb (hunk 2 above)
Comment (ademibowale) [reply spec/intergrationTests/post_show_page_spec.rb]: Thank you i will resolve that immidiately as you mentioned

File: spec/intergrationTests/post_index_page_spec.rb (hunk 1 above)
Comment (ademibowale) [reply spec/intergrationTests/post_index_page_spec.rb]: Thank so much great job and kudus to you

File: spec/features/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      end
+    end
+
+    it 'should redirect to user\'s show page' do
+      @users.each do |user|
+        find_link(user.Name).click
+        expect(page).to have_current_path(user_path(user))
+      end
+    end
+  end
```
Comment (AnselemOdims) [inline spec/features/user_spec.rb:34]: - [ ] As suggested by the previous reviewer, 
- please kindly include a test to assert that a user can see the number of likes a post has.
- please also include a test that asserts that you can see the username of each commenter

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @ademibowale, 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/features/user_spec.rb (hunk 4 above)
Comment (ademibowale) [reply spec/features/user_spec.rb:34]: @AnselemOdims if you check the project requirement, you'll see i have implemented all the necessary testing i need to do, you just came up with what is not in the requirement, and with what the previous reviewer asked me to do i have done that and yours does not go in line with what the previous reviewer asked me to do, 
Kindly check project requirement for peruse before you request for changes 
Thanks.

**Review by KDlamini — APPROVED** [review]
### Hi @ademibowale 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ Good PR description
✔️ integration specs for views well-implemented
✔️ No linter errors

# Status: Approved ✔️ 

LeonardoInGatsby

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**